azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessagingFactory.java

```diff
@@ -210,6 +216,8 @@ public void onOpenComplete(Exception exception)
 		{
 			this.open.complete(this);
 			this.openConnection.complete(this.connection);
+                        if (this.getIsClosingOrClosed())
+                            this.connection.close();
```

Comment:
Yes, we should & We already Do.
"Close-Connection" is a reactor event - which will need to flown to service. Once, client gets close ack from service, reactor will invoke onError(null) - which is equivalent to remoteClose - & in the OnError(null) - we stop the reactor.

---
In reply to: [109542770](https://github.com/Azure/azure-event-hubs-java/pull/88#discussion_r109542770) [](ancestors = 109542770)